### PR TITLE
Upgrade web-vitals vendor scripts to version 2.1.0

### PR DESCRIPTION
**Pull request by samuel-holt**
## Overview

Upgrade the [web-vitals](https://github.com/GoogleChrome/web-vitals) vendor scripts to the latest version ([v2.1.0](https://github.com/GoogleChrome/web-vitals/releases/tag/v2.1.0)). 

This version includes several updates to the library, including the following notable improvements:

- [Firefox compatibility improvements](https://github.com/GoogleChrome/web-vitals/pull/143/files)
- [CLS tracking improvement](https://github.com/GoogleChrome/web-vitals/pull/149/files) 
- [Only track CLS if FCP is being tracked](https://github.com/GoogleChrome/web-vitals/pull/154/files)

### Scripts updated

- [web-vitals-polyfills.vendor.js](https://github.com/MindscapeHQ/raygun4js/blob/3a4b0bd93359552a83b94112120534d92578af9b/src/raygun.rum/vendor/web-vitals-polyfills.vendor.js)
- [web-vitals.vendor.js](https://github.com/MindscapeHQ/raygun4js/blob/3a4b0bd93359552a83b94112120534d92578af9b/src/raygun.rum/vendor/web-vitals.vendor.js)

### CODE REVIEW

File: src/raygun.rum/vendor/web-vitals.vendor.js [hunk 1]
```diff
@@ -1,567 +1,303 @@
...
-        name: name,
-        value: typeof value === 'undefined' ? -1 : value,
-        delta: 0,
-        entries: [],
-        id: generateUniqueID()
-      };
+(function(exports) {
+  'use strict';
+
+  // This ensures that we do not initialize Core Web Vitals for non-browser environments
```
Comment (mduncan26) [inline src/raygun.rum/vendor/web-vitals.vendor.js:9]: This is is an update we added to this script? Do we want to make it more clear that we added this block by tagging things? Not a necessity.

E.g. "Raygun: This  ensures that we ..."

File: src/raygun.rum/vendor/web-vitals-polyfills.vendor.js [hunk 2]
```diff
@@ -1,232 +1,106 @@
...
-
-    var firstInputPolyfill = function firstInputPolyfill(onFirstInput) {
-      callbacks.push(onFirstInput);
+/**
+ * web-vitals v2.1.0 polyfill
+ * This comes from the google web-vital repository, polyfill script @https://github.com/GoogleChrome/web-vitals
+ */
+
+(function() {
+  // This ensures that we do not initialize Core Web Vitals for non-browser environments
```
Comment (mduncan26) [inline src/raygun.rum/vendor/web-vitals-polyfills.vendor.js:7]: This is is an update we added to this script? Do we want to make it more clear that we added this block by tagging things? Not a necessity.

E.g. "Raygun: This ensures that we ..."

**Comment by mduncan26** [comment]
Are you able to confirm that all metrics will be reported from these scripts after the page onload event.

**Comment by samuel-holt** [comment]
> Are you able to confirm that all metrics will be reported from these scripts after the page onload event.

Yes, the core web vital metrics can technically be tracked before page load by the web-vitals library, **but** Raygun4JS doesn't attach handlers to the getLCP, getFID and getCLS functions until after the page has [loaded](https://github.com/MindscapeHQ/raygun4js/blob/master/src/raygun.loader.js#L241).

**Review by mduncan26 — APPROVED** [review]
🌮

**Review by mduncan26 — APPROVED** [review]
🌮

**Review by mduncan26 — APPROVED** [review]

**Pull request merged**